### PR TITLE
Added 'task_summarize_all_person' function to summary.py

### PR DESCRIPTION
**Pull request by kn2322**
To run 'task_summarize_all_person', download person.json from summary issue, and place it in local/person/. The new function uses desirable properties from the Person, and generates a short description from a template.

### CODE REVIEW

File: cdata/summary.py [hunk 1]
```diff
@@ -27,17 +29,64 @@ def summarize_entity_person(person):
...
+        ret.append(u'主要作品有{}'.format(value))
+
     value = person.get("accomplishment")
     if value and len(value) < 30:
         ret.append( u"主要成就是{}".format(value) )
 
-    return u"，".join(ret)
+    ret = u"，".join(ret)
+    if ret[-1] != u'。':
+        ret += u'。'
```
Comment (lidingpku) [inline cdata/summary.py]: sub (“[，]+”,“，”）

File: cdata/summary.py [hunk 2]
```diff
@@ -49,8 +98,8 @@ def task_summarize_entity_person(args):
     logging.info(ret)
 
 def task_summarize_all_person(args):
-    person_list = []
-    #TODO load person_list from json
+    path2person = file2abspath('../local/person/person.json')
+    person_list = (json.loads(i) for i in file2iter(path2person))
```
Comment (lidingpku) [inline cdata/summary.py]: optimize

File: cdata/summary.py [hunk 3]
```diff
@@ -61,6 +110,11 @@ def task_summarize_all_person(args):
 
     logging.info( "write to JSON and excel")
 
+    KEYS = [u"@type",u"artName",u"ethnicGroup",u"student",u"courtesyName",u"religion",u"cnProfessionalTitle",u"occupation",u"jobTitle",u"sibling",u"weight",u"nationality",u"birthPlace",u"height",u"alumniOf",u"keywords", u"schoolsOfbuddhism",u"image",u"parent",u"children",u"accomplishment",u"academicDegree",u"dharmaName",u"deathDate",u"academicMajor",u"nobleTitle",u"posthumousName",u"familyName",u"memberOfPoliticalParty",u"award",u"description", u"shortDescription", u"placeOfBurial",u"cnEducationalAttainment",u"alternateName",u"pseudonym",u"templeName",	u"birthDate",u"gender",u"worksFor",u"name",u"dynasty",u"earName",u"ancestralHome",u"birthName",u"studentOf",u"spouse",u"nobleFamily",u"authorOf",u"@id",u"colleague",u"fieldOfWork",u"mother",u"father"]
+
+    json2excel(result_person_list, KEYS, 'person_shortDescription.xls')
```
Comment (lidingpku) [inline cdata/summary.py]: local/person

File: tests/test_summary.py [hunk 4]
```diff
@@ -23,10 +23,7 @@ def test_misc(self):
             "accomplishment": u"三好学生"
         }
         ret = summarize_entity_person(person)
-        assert u"张三，主要成就是三好学生" == ret, ret
-
-
-
+        assert u"张三，主要成就是三好学生。" == ret
```
Comment (lidingpku) [inline tests/test_summary.py]: more test

**Comment by kn2322** [comment]
Addressed issues shown in code review

**Comment by kn2322** [comment]
Fixed minor issues after testing with data for well-known people.

File: cdata/summary.py [hunk 5]
```diff
@@ -27,17 +29,74 @@ def summarize_entity_person(person):
...
+
+    ret = u"，".join(ret)
+
+    # Make all commas Chinese
+    ret = ret.replace(u',', u'，')
+    ret = re.sub(u"，+", u"，", ret) # Removes repeat commas
+    # Handles periods at end
+    ret = re.sub(u"[。，]+$", u"", ret)
+    # Removes brackets
+    ret = re.sub(ur"\（[\s\S]*\）", u"", ret)
```
Comment (lidingpku) [inline cdata/summary.py]: [\s\S]  =>  [^）]

File: cdata/summary.py [hunk 6]
```diff
@@ -27,17 +29,74 @@ def summarize_entity_person(person):
...
+
+    ret = ''.join([ret, u"。"])
```
Comment (lidingpku) [inline cdata/summary.py]: u''.join()

ret + u"。"

File: cdata/summary.py [hunk 7]
```diff
@@ -27,17 +29,74 @@ def summarize_entity_person(person):
...
+        # Colon is handled by text reading software
+        ret.append( u"主要成就：{}".format(value) )
+
+    ret = u"，".join(ret)
+
+    # Make all commas Chinese
+    ret = ret.replace(u',', u'，')
+    ret = re.sub(u"，+", u"，", ret) # Removes repeat commas
+    # Handles periods at end
+    ret = re.sub(u"[。，]+$", u"", ret)
```
Comment (lidingpku) [inline cdata/summary.py]: ur"...."

File: cdata/summary.py [hunk 8]
```diff
@@ -27,17 +27,78 @@ def summarize_entity_person(person):
...
+    ret = ret.replace(u',', u'，')
+    ret = re.sub(u"，+", u"，", ret) # Removes repeat commas
+    # Handles periods at end
+    ret = re.sub(ur"[。，]+$", u"", ret)
+
+    # Converts brackets to Chinese
+    ret = ret.replace(u'(', u'（')
+    ret = ret.replace(u')', u'）')
+    # Removes brackets and all contained info
+    ret = re.sub(ur"\（[^）]*\）", u"", ret)
```
Comment (lidingpku) [inline cdata/summary.py]: 不用转义

**Review by lidingpku — APPROVED** [review]

**Pull request merged**